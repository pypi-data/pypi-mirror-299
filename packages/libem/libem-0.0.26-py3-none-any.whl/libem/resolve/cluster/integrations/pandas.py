import pandas as pd

from libem.resolve.cluster.function import func as cluster_func

schema = {
    "type": "function",
    "function": {
        "name": "cluster",
        "description": "Perform the clustering stage of "
                       "entity matching given a dataframe "
                       "and add a __cluster__ column.",
        "parameters": {
            "type": "object",
            "properties": {
                "df": {
                    "type": "dataframe",
                    "description": "A pandas DataFrame containing the dataset.",
                },
            },
            "required": ["df"],
        },
    }
}


def cluster(*args, **kwargs):
    return func(*args, **kwargs)


def func(df: pd.DataFrame, sort: bool = False) -> pd.DataFrame:
    clusters = cluster_func(
        df.to_dict(orient="records")
    )

    new_df = df.copy()
    new_df["__cluster__"] = [cluster_id for cluster_id, _ in clusters]

    if sort:
        return new_df.sort_values(by="__cluster__")

    return new_df
