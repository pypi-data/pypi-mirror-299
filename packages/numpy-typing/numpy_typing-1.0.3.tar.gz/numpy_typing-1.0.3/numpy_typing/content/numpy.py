"""Numpy overload with type hinting"""

from . import typing as np
from typing import overload, TypeVar, Any


T = TypeVar('T')
T1 = TypeVar('T1')
T2 = TypeVar('T2')
T3 = TypeVar('T3')
T4 = TypeVar('T4')


# |====================================================================================================================
# | Numpy functions
# |====================================================================================================================


def concatenate(arrays: "list[T]|tuple[T,...]",
                axis: int = None,
                out: T = None, *,
                dtype:Any = None,
                casting:str = None) -> T: ...



def abs(a: T) -> T: ...
def add(a: T, b: T, /) -> T: ...



from numpy import *

max = amax
min = amin
abs = absolute



