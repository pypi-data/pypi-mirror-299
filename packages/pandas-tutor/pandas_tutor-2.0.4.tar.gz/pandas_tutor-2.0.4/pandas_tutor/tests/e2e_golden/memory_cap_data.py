import pandas as pd
import numpy as np

# this uses about 1.5 MB
big = pd.DataFrame({'numbers': np.arange(200_000)})

big.head()
