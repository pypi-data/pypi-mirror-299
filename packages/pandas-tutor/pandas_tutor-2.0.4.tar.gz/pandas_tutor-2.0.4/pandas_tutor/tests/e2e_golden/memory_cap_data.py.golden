{
  "code": "import pandas as pd\nimport numpy as np\n\n# this uses about 1.5 MB\nbig = pd.DataFrame({'numbers': np.arange(200_000)})\n\nbig.head()\n",
  "explanation": [
    {
      "type": "HeadCall",
      "code_step": "big.head()",
      "fragment": {
        "start": {
          "line": 0,
          "ch": 3
        },
        "end": {
          "line": 0,
          "ch": 10
        }
      },
      "marks": [
        {
          "type": "map",
          "from": {
            "type": "axis",
            "anchor": "lhs",
            "select": "row",
            "label": 0
          },
          "to": {
            "type": "axis",
            "anchor": "rhs",
            "select": "row",
            "label": 0
          }
        },
        {
          "type": "map",
          "from": {
            "type": "axis",
            "anchor": "lhs",
            "select": "row",
            "label": 1
          },
          "to": {
            "type": "axis",
            "anchor": "rhs",
            "select": "row",
            "label": 1
          }
        },
        {
          "type": "map",
          "from": {
            "type": "axis",
            "anchor": "lhs",
            "select": "row",
            "label": 2
          },
          "to": {
            "type": "axis",
            "anchor": "rhs",
            "select": "row",
            "label": 2
          }
        },
        {
          "type": "map",
          "from": {
            "type": "axis",
            "anchor": "lhs",
            "select": "row",
            "label": 3
          },
          "to": {
            "type": "axis",
            "anchor": "rhs",
            "select": "row",
            "label": 3
          }
        },
        {
          "type": "map",
          "from": {
            "type": "axis",
            "anchor": "lhs",
            "select": "row",
            "label": 4
          },
          "to": {
            "type": "axis",
            "anchor": "rhs",
            "select": "row",
            "label": 4
          }
        }
      ],
      "data": {
        "lhs": {
          "type": "DataFrame",
          "columns": {
            "names": [
              null
            ],
            "labels": [
              "numbers"
            ]
          },
          "index": {
            "names": [
              null
            ],
            "labels": [
              0,
              1,
              2,
              3,
              4,
              5,
              6,
              7,
              8,
              9,
              10,
              11,
              12,
              13,
              14,
              15,
              16,
              17,
              18,
              19,
              20,
              21,
              22,
              23,
              24,
              199975,
              199976,
              199977,
              199978,
              199979,
              199980,
              199981,
              199982,
              199983,
              199984,
              199985,
              199986,
              199987,
              199988,
              199989,
              199990,
              199991,
              199992,
              199993,
              199994,
              199995,
              199996,
              199997,
              199998,
              199999
            ]
          },
          "data": [
            [
              0
            ],
            [
              1
            ],
            [
              2
            ],
            [
              3
            ],
            [
              4
            ],
            [
              5
            ],
            [
              6
            ],
            [
              7
            ],
            [
              8
            ],
            [
              9
            ],
            [
              10
            ],
            [
              11
            ],
            [
              12
            ],
            [
              13
            ],
            [
              14
            ],
            [
              15
            ],
            [
              16
            ],
            [
              17
            ],
            [
              18
            ],
            [
              19
            ],
            [
              20
            ],
            [
              21
            ],
            [
              22
            ],
            [
              23
            ],
            [
              24
            ],
            [
              199975
            ],
            [
              199976
            ],
            [
              199977
            ],
            [
              199978
            ],
            [
              199979
            ],
            [
              199980
            ],
            [
              199981
            ],
            [
              199982
            ],
            [
              199983
            ],
            [
              199984
            ],
            [
              199985
            ],
            [
              199986
            ],
            [
              199987
            ],
            [
              199988
            ],
            [
              199989
            ],
            [
              199990
            ],
            [
              199991
            ],
            [
              199992
            ],
            [
              199993
            ],
            [
              199994
            ],
            [
              199995
            ],
            [
              199996
            ],
            [
              199997
            ],
            [
              199998
            ],
            [
              199999
            ]
          ]
        },
        "rhs": {
          "type": "DataFrame",
          "columns": {
            "names": [
              null
            ],
            "labels": [
              "numbers"
            ]
          },
          "index": {
            "names": [
              null
            ],
            "labels": [
              0,
              1,
              2,
              3,
              4
            ]
          },
          "data": [
            [
              0
            ],
            [
              1
            ],
            [
              2
            ],
            [
              3
            ],
            [
              4
            ]
          ]
        }
      }
    }
  ]
}